traitsui/tests/editors/test_image_enum_editor.py

```diff
@@ -0,0 +1,488 @@
+                editor=ImageEnumEditor(
+                    name="object.possible_values",
+                    format_func=lambda v: v.upper(),
+                    prefix='@icons:',
+                    suffix='_origin',
+                    path='dummy_path',
+                ),
+                style="readonly",
+            )
+        )
+        model = PossibleEnumModel()
+
+        with store_exceptions_on_all_threads():
+            editor = self.setup_ui(model, formatted_view)
+
+            # Readonly editor doesn't set up full mapping, only check that
+            # str_value is mapped as expected
+            self.assertEqual(model.value, "top left")
+            self.assertEqual(editor.str_value, "TOP LEFT")
+
+
+@skip_if_null
+class TestSimpleImageEnumEditor(unittest.TestCase):
+
+    def setup_gui(self, model, view):
+        gui = GUI()
+        ui = model.edit_traits(view=view)
+        self.addCleanup(ui.dispose)
+
+        gui.process_events()
+        editor = ui.get_editors("value")[0]
+        control = editor.control
+
+        return gui, control
+
+    @unittest.skipIf(is_linux, "Issue enthought/traitsui#854")
+    def test_simple_editor_more_cols(self):
+        # Smoke test for setting up an editor with more than one column
+        enum_edit = EnumModel()
+        view = View(
+            UItem(
+                'value',
+                editor=ImageEnumEditor(
+                    values=[
+                        'top left', 'top right', 'bottom left', 'bottom right'
+                    ],
+                    prefix='@icons:',
+                    suffix='_origin',
+                    path='dummy_path',
+                    cols=4,
+                ),
+                style="simple",
+            ),
+            resizable=True,
+        )
+
+        with store_exceptions_on_all_threads():
+            self.setup_gui(enum_edit, view)
+
+    @skip_if_not_wx
+    def test_simple_editor_popup_editor(self):
+        enum_edit = EnumModel()
+
+        with store_exceptions_on_all_threads():
+            gui, control = self.setup_gui(enum_edit, get_view("simple"))
+
+            self.assertEqual(enum_edit.value, 'top left')
+
+            # Set up ImageEnumDialog
+            click_on_image(control)
+            gui.process_events()
+
+            # Check created buttons
+            image_grid_control = control.GetChildren()[0].GetChildren()[0]
+            self.assertEqual(
+                get_button_strings(image_grid_control),
+                ['top left', 'top right', 'bottom left', 'bottom right']
+            )
+
+            # Select new image
+            click_on_image(get_button_control(image_grid_control, 1))
+            gui.process_events()
+
+            self.assertEqual(enum_edit.value, 'top right')
+
+            # Check that dialog window is closed
+            self.assertEqual(list(control.GetChildren()), [])
+
+    @skip_if_not_qt4
+    @unittest.skipIf(is_linux, "Issue enthought/traitsui#854")
```

Comment:
Maybe mention "Possible test interaction on Linux"?  This would indicate that the test should pass while run on its own. One is more likely to try running this test if they see the reason from their console without having to dive into the issue.